components/Form/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,155 @@
+import { forwardRef } from 'react';
+import Downshift from 'downshift';
+import cx from 'classnames';
+import { Controller, Control } from 'react-hook-form';
+import { sortObject } from 'utils/objects';
+import AutoInput from './AutoInput';
+
+import {
+  OptionObject,
+  Autocomplete as IAutocomplete,
+} from 'components/Form/types';
+
+interface AutoProps extends IAutocomplete {
+  ref?: () => void;
+  options?: OptionObject[];
+  onChange: (arg0: string | null) => void;
+  value?: string | null;
+  inputClassName?: string;
+}
+
+interface Props extends IAutocomplete {
+  control: Control;
+  onChange: (arg0: string | null) => void;
+  value?: string;
+}
+
+// eslint-disable-next-line react/display-name
+export const Autocomplete = forwardRef<HTMLInputElement, AutoProps>(
+  (
+    {
+      label,
+      options,
+      hint,
+      name,
+      error,
+      labelSize = 'm',
+      required,
+      onChange,
+      placeholder,
+      width = 20,
+      govGrid,
+      value,
+      inputClassName,
+      ...otherProps
+    }: AutoProps,
+    ref
+  ) => {
+    const optionObject = options?.map((option: OptionObject) => {
+      if (typeof option === 'string') {
+        return { value: option, text: option };
+      }
+      return option;
+    });
+    const items = sortObject(optionObject);
```

Comment:
I think we can use `useMemo` so we don't need to parse and sort all the options on change